### PR TITLE
migrate from drone to gh actions

### PR DESCRIPTION
**Pull request by slickwarren**
#### What this PR does / why we need it:
/kind cleanup
This PR migrates CI and release from Drone to GH actions

#### Special notes for your reviewer:
* Releases [will look like ths](https://github.com/slickwarren/rancher-kubernetes/releases)
* Actions will [look like this](https://github.com/slickwarren/rancher-kubernetes/actions)
  * note that Alphas/RCs will fail to release (in Drone, these were skipped)
  * Releases have 2 stages, `validate` where CI runs, and `create-release` which releases the tagged version
  * CI runs on each PR + each merge to the target branch
* [Drone of a release](https://drone-publish.rancher.io/rancher/kubernetes/256/1/3) to compare with the actions above
* upon initial approval to the default branch (release-1.30) I will open ports to the other current versions (release-1.29, release-1.28, release-1.27)

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+          overwrite: true
+
+  create-release:
+    runs-on: ubuntu-latest
+    needs:
+      - validate
+    permissions:
+      contents: write # needed for creating the GH release
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (jiaqiluo) [inline .github/workflows/release.yml:54]: GH_TOKEN is the recommended env var to use according to https://docs.github.com/en/actions/using-workflows/using-github-cli-in-workflows
```suggestion
      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      GIT_SHA: sha256
```
Comment (jiaqiluo) [inline .github/workflows/release.yml:55]: Where is it used?

**Review by jiaqiluo — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (slickwarren) [reply .github/workflows/release.yml]: updated to GH_TOKEN

File: .github/workflows/release.yml (hunk 2 above)
Comment (slickwarren) [reply .github/workflows/release.yml]: removed the extra env

**Review by jiaqiluo — APPROVED** [review]

**Pull request merged**